### PR TITLE
add concentration of samples in reactions (merge Mass/Vol) resolve #534 #535 #536

### LINKED ISSUE #534: Private note rework
Can we minimize the field "private note" field? its quite large and only used in certain cases. Also, it would be good to remove the save button if possible (the only field with additional save function)

### PR DESCRIPTION
**Pull request by JasonYCHuang**
merge Mass/Vol to get more place for concentration.

### CODE REVIEW

**Comment by Sergeykot** [comment]
nan1

concentration - NaN when adding a reaction starting material

**Comment by Sergeykot** [comment]
I am testing it myself, and notice that now reactions yield is calculated when user unfocuses input field
before it was on keyboard event

I have nothing against if those changes were requested :)

File: app/assets/javascripts/components/Material.js [hunk 1]
```diff
@@ -248,9 +261,12 @@ class Material extends Component {
 
   generalMaterial(props, style, handleStyle, inputsStyle) {
     const {material, deleteMaterial, isDragging, connectDragSource,
-           showLoadingColumn } = props;
+           showLoadingColumn, totalVolume } = props;
     const isTarget = material.amountType === 'target'
 
+    const mol = material.amount_mol;
+    const concentration = mol / totalVolume;
```
Comment (Sergeykot) [inline app/assets/javascripts/components/Material.js]: totalVolume should not be 0 otherwise we get `NaN`

Comment (JasonYCHuang) [inline app/assets/javascripts/components/Material.js]: Done. plz check it again.

**Comment by Sergeykot** [comment]
:+1:

**Pull request closed without merging**